packages/web/hooks/use-estimate-tx-fees.ts

```diff
@@ -28,34 +28,34 @@ async function estimateTxFeesQueryFn({
 }) {
   if (!messages) throw new Error("No messages");
 
-  const [{ amount, gas: gasLimit }, osmoAssetWithPrice] = await Promise.all([
-    accountStore.estimateFee({
-      wallet,
-      messages: messages!,
-      signOptions: {
-        ...wallet.walletInfo?.signOptions,
-        preferNoSetFee: true, // this will automatically calculate the amount as well.
-      },
-    }),
-    apiUtils.edge.assets.getMarketAsset.fetch({
-      findMinDenomOrSymbol: "OSMO",
-    }),
-  ]);
+  const { amount, gas: gasLimit } = await accountStore.estimateFee({
+    wallet,
+    messages: messages!,
+    signOptions: {
+      ...wallet.walletInfo?.signOptions,
+      preferNoSetFee: true, // this will automatically calculate the amount as well.
+    },
+  });
 
   const coin = amount[0];
-  if (!coin || !osmoAssetWithPrice?.currentPrice) {
+
+  const asset = await apiUtils.edge.assets.getMarketAsset.fetch({
+    findMinDenomOrSymbol: coin.denom,
+  });
```

Comment:
Using `getAssetPrice` may be faster since `getMarketData` fetches a lot of other info along with the price.

Reply:
I need the decimals so I created a new query `getAssetWithPrice` that includes the asset with its price.